tools/gen_stubs.zig

```diff
@@ -0,0 +1,31 @@
+const std = @import("std");
+
+// Example usage:
+// objdump --dynamic-syms /path/to/libc.so | ./gen_stubs > lib/libc/musl/libc.s
+pub fn main() !void {
+    const stdin = std.io.getStdIn().reader();
+    const stdout = std.io.getStdOut().writer();
+
+    // This is many times larger than any line objdump produces should ever be
+    var buf: [4096]u8 = undefined;
+
+    // Sample input line:
+    // 00000000000241b0 g    DF .text	000000000000001b copy_file_range
```

Comment:
If you want to do some more filtering check out glibc's `abilist.awk` (see [here](https://raw.githubusercontent.com/bminor/glibc/master/scripts/abilist.awk]). Musl's `libc.so` is quite tame if compared to glibc's one so most of that is not needed (yet?)/


One thing that's wrong even in the glibc parsing scripts is the use of `.type N, @function;`, that's ok for functions but not really ok for other symbol kinds (eg. `environ` is marked as `DO` instead of `DF` in objdump's output).

Since the symbol types "percolate" to the final object this may be a problem...

Here's the objdump's output of a simple program referencing `environ`, the first part is when musl's libc is used, the second when the generated `libc.so` is used instead:

```
     5: 0000000000004028     8 OBJECT  WEAK   DEFAULT   20 _environ
     6: 0000000000004028     8 OBJECT  WEAK   DEFAULT   20 environ
     7: 0000000000004028     8 OBJECT  GLOBAL DEFAULT   20 __environ
     9: 0000000000004028     8 OBJECT  WEAK   DEFAULT   20 ___environ
    48: 0000000000004028     8 OBJECT  GLOBAL DEFAULT   20 __environ
    50: 0000000000004028     8 OBJECT  WEAK   DEFAULT   20 environ
```

```
   523: 0000000000001016     0 FUNC    GLOBAL DEFAULT    6 _environ
   552: 0000000000001016     0 FUNC    GLOBAL DEFAULT    6 __environ
   703: 0000000000001016     0 FUNC    GLOBAL DEFAULT    6 environ
   820: 0000000000001016     0 FUNC    GLOBAL DEFAULT    6 ___environ
```

Will this be a problem? In theory yes, every tool that checks the symbol type will be extremely confused by this.